pkg/auth/jwt.go

```diff
@@ -0,0 +1,68 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+
+	"NKSS-backend/pkg/handlers"
+
+	"github.com/golang-jwt/jwt"
+	_ "github.com/lib/pq"
+)
+
+func Authenticator(next http.Handler, secret string) http.Handler {
```

Comment:
Why not just return a `HandlerFunc` ?

https://github.com/NIT-KKR-Student-Support-System/nkss-backend/blob/73e2c9a8b1476ced1c5526bdaee0195d9642b673/pkg/app.go#L47